src/pages/index.js

```diff
@@ -20,100 +24,178 @@ import {
 /*---------------------------------------------------*/
 
 /*-- Add Modal --*/
-const addModal = document.querySelector("#add-modal");
 const addModalForm = document.querySelector("#add-modal-form");
-const addModalTitle = addModalForm.querySelector("#add-modal-title");
-const addModalURL = addModalForm.querySelector("#add-modal-url");
-const addModalCloseButton = addModal.querySelector(".modal__close");
-
-/*-- Card Selector --*/
-const cardTemplate =
-  document.querySelector("#card-template").content.firstElementChild;
-const cardElement = cardTemplate.cloneNode(true);
-const cardListEl = document.querySelector(".card__list");
-const cardImageEl = cardElement.querySelectorAll(".card__image");
+const addModalSubmitButton = addModalForm.querySelector(".modal__button");
 
 /*-- Edit Modal --*/
 const editModal = document.querySelector("#edit-modal");
 const editModalForm = editModal.querySelector("#edit-modal-form");
-const editModalName = editModal.querySelector("#edit-modal-name");
-const editModalBio = editModal.querySelector("#edit-modal-bio");
-const editModalCloseButton = editModal.querySelector(".modal__close");
-
-/*-- Preview Modal --*/
-const previewModal = document.querySelector("#preview-modal");
-const previewModalCloseButton = document.querySelector("#preview-modal-button");
-const previewTitleEl = previewModal.querySelector("#modal-preview-title");
-const previewImageEl = previewModal.querySelector(".modal__preview-image");
+const editModalSubmitButton = editModalForm.querySelector(".modal__button");
 
 /*-- Profile Selectors --*/
+const profileAvatarForm = document.querySelector("#profile-modal-form");
+const profilePenIcon = document.querySelector(`#profile__avatar-pen-icon`);
 const profileEditButton = document.querySelector(".profile__button-edit");
 const profileAddButton = document.querySelector(".profile__button-add");
-const profileName = document.querySelector(".profile__name");
-const profileBio = document.querySelector(".profile__bio");
+const avatarModalSubmitButton =
+  profileAvatarForm.querySelector(".modal__button");
 
-const items = cardsData;
-const section = new Section({ items, renderer: createCard }, ".card__list");
-section.renderItems();
+/*-- Trash Icon Selectors --*/
+const trashModal = document.querySelector("#trash-modal");
+const trashModalSubmitButton = trashModal.querySelector(".modal__button");
 
 /*---------------------------------------------------*/
-/*                     Functions                     */
+/*                 Event Handlers                    */
 /*---------------------------------------------------*/
 
-function createCard(cardData) {
-  const name = cardData.name;
-  const altName = cardData.name;
-  const link = cardData.link;
-  const card = new Card(
-    { name, altName, link },
-    "#card-template",
-    handleImageClick
-  );
-  return card.getView();
+function handleAvatarFormSubmit(formValues) {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```